web_external/js/views/body/ImagesView.js

```diff
@@ -0,0 +1,370 @@
+          result.minIndices = [index];
+        } else if (column.length === result.minLength) {
+          result.minIndices.push(index);
+        }
+      });
+
+      // Choose the single min and max columns randomly
+      result.min = Math.floor(Math.random() * result.minIndices.length);
+      result.min = result.minIndices[result.min];
+      result.max = Math.floor(Math.random() * result.maxIndices.length);
+      result.max = result.maxIndices[result.max];
+
+      return result;
+    }
+
+    // Consume imagesToPlace by putting images
+    // in one of the least-full columns
+    while (imagesToPlace.length > 0) {
+      imageId = imagesToPlace.pop();
+      myColumn = minAndMaxIndices().min;
+
+      self.imageColumns[myColumn].push(imageId);
+      self.imageColumnLookup[imageId] = myColumn;
+    }
+
+    // Balance the columns
+    var minAndMax = minAndMaxIndices();
+    while (minAndMax.maxLength - minAndMax.minLength > 3) {
+      // Allow a ragged bottom edge... the gap can be as much as 3
+      imageId = self.imageColumns[minAndMax.max].pop();
+      self.imageColumns[minAndMax.min].push(imageId);
+      self.imageColumnLookup[imageId] = minAndMax.min;
+
+      minAndMax = minAndMaxIndices();
+    }
+  },
+  renderImages: function () {
+    var self = this;
+
+    // Figure out how much space we have to play with
+    var width = widthWithoutScrollbar('#pageContent');
+
+    // We want the width of each column of images to have an
+    // inverse relationship with the size of the download, so
+    // more, smaller thumbnails show up when the download is large)
+    var columnScale = d3.scale.linear()
+      .domain([0, temp_download_size])
+      .range([256, 256 / 4]); // Shrink the thumbnails by as much as a quarter
+
+    var imageWidth = columnScale(temp_download_size);
+
+    var imagePadding = 5;
+
+    // We'll do a little sneaky scaling up so the images
+    // are flush with the edge
+    var numColumns = Math.floor((width - imagePadding) /
+      (imageWidth + imagePadding));
+    imageWidth = (width - imagePadding) / numColumns - imagePadding;
+
+    // Okay, we know how many columns we can fit...
+    // figure out where all the images go:
+    self.rearrangeColumns(numColumns);
+
+    // Okay, time to draw / move the pictures:
+    var images = d3.select('#preview').selectAll('image')
+      .data(self.imageIds, function (d) {
+        return d;
+      });
+    images.enter().append('image')
+      .attr({
+        'id': function (d) {
+          return 'image' + d;
+        },
+        'xlink:href': function (d) {
+          return self.imageCache[d].src;
+        },
+        'preserveAspectRatio': 'xMinYMin',
+        'x': width / 2, // Start new images at the middle of the
+        'y': jQuery('#preview').height() // bottom of the screen
+      })
+      .on('click', function (d) {
+        // TODO: show a lightbox
+        // For now, jump to the item in girder
+        window.open('#item/' + d);
```

Comment:
Let's make this point at `/girder#item/`, now that https://github.com/ImageMarkup/isic-archive/pull/99 is merged.